### PR TITLE
refactor(s3): Move all cache files to the `cache/` folder

### PR DESCRIPTION
**Pull request by ViBiOh**
### What does this PR do?

Change the location of caches files under a `cache/` folder prefix. So that it's allowed by the Policy and will result in a NotFoundException instead of AccessDeniedException, as explained [here](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html).

### Motivation

avoid CloudTrail pollution for AccessDeniedException

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [x] This PR passes the integration tests (ask a Datadog member to run the tests)
- [x] This PR passes the unit tests 
- [x] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: aws/logs_monitoring/lambda_function.py [hunk 1]
```diff
@@ -4,29 +4,30 @@
 # Copyright 2021 Datadog, Inc.
 
 import json
+import logging
```
Comment (ViBiOh) [inline aws/logs_monitoring/lambda_function.py]: Just formatting when passing by.

File: aws/logs_monitoring/template.yaml [hunk 2]
```diff
@@ -794,14 +807,6 @@ Resources:
...
                     Effect: Allow
                   - Ref: AWS::NoValue
-              # Fetch Step Functions resource tags for data enrichment
-              - Fn::If:
-                  - SetDdFetchStepFunctionsTags
-                  - Action:
-                      - tag:GetResources
-                    Resource: "*"
-                    Effect: Allow
-                  - Ref: AWS::NoValue
```
Comment (ViBiOh) [inline aws/logs_monitoring/template.yaml]: This was a duplicate I fixed along the way.

**Review by tyrcho — APPROVED** [review]

**Comment by klivan** [comment]
My concern is that `List*` operations are 10x more expensive than `Get*`, so we'll 10x customers' costs for these checks.

**Comment by tyrcho** [comment]
> My concern is that `List*` operations are 10x more expensive than `Get*`, so we'll 10x customers' costs for these checks.

If this cost becomes a concern, I guess we could offer both alternatives, configurable with an env variable. 
By default we should keep the existing behaviour, and if the variable is set we would switch to List operations to reduce CloudTrail pollution with some extra S3 cost.

**Pull request merged**